src/Kestrel.Core/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -90,8 +90,21 @@ string IHttpRequestFeature.Scheme
 
         string IHttpRequestFeature.Method
         {
-            get => Method;
-            set => Method = value;
+            get
+            {
+                if (_customMethod != null)
+                {
+                    return _customMethod;
+                }
+
+                _customMethod = HttpUtilities.MethodToString(Method);
```

Comment:
The old code did `HttpUtilities.MethodToString(method) ?? string.Empty`. I'd either keep that or have MethodToString throw instead of return null neither of which should ever actually happen in practice.

Reply:
done